### DISCUSSION TITLE
How to focus on previous work space if current space is destoryed?

### DISCUSSION THREAD

**Discussion by aurexav** [asmvik/yabai#1095] [Q&A] [answered]
```
yabai -m space "$(yabai -m query --spaces --space | jq '.index')" --destroy
```

Currently, it will always focus on index 1 once the current space is destroyed.

**Comment by aurexav** [comment, marked as answer]
I came out with this

```
space="$(yabai -m query --spaces --space | jq '.index')" && \
         prev_space=$(($space - 1)) && \
         yabai -m space --focus "${prev_space}" && \
         yabai -m space "${space}" --destroy
```

But if my previous work space is 1, current work space is 3, this will lead me to the space 2.

---

```
yabai -m space --focus recent && sleep 1 && yabai -m space recent --destroy
```

This works for me. It seems the status has not changed if too fast.

**Reply by asmvik** [reply]
See https://github.com/koekeishiya/yabai/discussions/964

**Reply by aurexav** [reply]
Oh! Sorry for the duplication!
